frame/staking/src/pallet/impls.rs

```diff
@@ -1405,17 +1405,17 @@ impl<T: Config> StakingInterface for Pallet<T> {
 		Self::unbond(RawOrigin::Signed(controller).into(), value)
 	}
 
+	fn chill(controller: Self::AccountId) -> DispatchResult {
+		Self::chill(RawOrigin::Signed(controller).into())
+	}
+
 	fn withdraw_unbonded(
 		controller: Self::AccountId,
 		num_slashing_spans: u32,
-	) -> Result<u64, DispatchError> {
-		Self::withdraw_unbonded(RawOrigin::Signed(controller).into(), num_slashing_spans)
-			.map(|post_info| {
-				post_info
-					.actual_weight
-					.unwrap_or(T::WeightInfo::withdraw_unbonded_kill(num_slashing_spans))
-			})
-			.map_err(|err_with_post_info| err_with_post_info.error)
+	) -> Result<bool, DispatchError> {
+		Self::withdraw_unbonded(RawOrigin::Signed(controller.clone()).into(), num_slashing_spans)
```

Comment:
I have no idea of this code.  
Not sure if @emostov is still reading here, otherwise you could approve?

Reply:
If you look at the current [code](https://github.com/paritytech/substrate/pull/11426/files#diff-59da7422e74f2a1b8bd5f6d0cbcc83fa5dda6a0b15a7995a333072742689402cL1633), `T::StakingInterface::withdraw_unbonded`'s return value is never used. This return value was supposed to be a weight value, and I am not sure how it was supposed to be used, but without it, the worse that can happen to us is that the weight here is a bit inaccurate, which I think is a fine compromise. 


Now I set it to something that is used to sanity check. The return value is signifying whether the stash/ledger was removed from storage as a result of calling this withdraw or not. This should only happen once a depositor withdraws.